apps/aeutil/lib/bits.ex

```diff
@@ -1,14 +1,20 @@
 # Taken from http://minhajuddin.com/2016/11/01/how-to-extract-bits-from-a-binary-in-elixir/
 # License: CC BY-SA 3.0
 defmodule Aeutil.Bits do
-  def bech32_encode(prefix, bin) do
-    SegwitAddr.encode(prefix, 0, :binary.bin_to_list(bin))
+  @prefix_list ["ak$", "bh$", "bs$", "tx$", "bx$", "ok$", "cs$", "tr$"]
+
+  def encode58(prefix, data) when is_binary(data) do
+    prefix <> Kernel.to_string(:base58.binary_to_base58(data))
   end
 
-  def bech32_decode(bech32) do
-    case SegwitAddr.decode(bech32) do
-      {:ok, {_, _, bin_list}} -> :binary.list_to_bin(bin_list)
-      {:error, _} = error -> error
+  def decode58(data) when is_binary(data) do
```

Comment:
can you pass the expected prefix as atom here as well, fail if the one in data doesn't match the expected prefix

Reply:
rename to `base58c_decode`

Reply:
that about this?